amqp/consumer.go

```diff
@@ -19,19 +19,66 @@ type Consumer struct {
 	conn     *Connection
 	autoAck  bool
 	handlers []handler
+
+	channel *amqplib.Channel
+	queue   *amqplib.Queue
+
+	exchangeName string
+	queueName    string
 }
 
 // NewConsumer returns a new AMQP Consumer.
-func NewConsumer(c messaging.Connection, autoAck bool) (messaging.Consumer, error) {
+func NewConsumer(c messaging.Connection, autoAck bool, exchange, queue string) (messaging.Consumer, error) {
 	amqpConn := c.(*Connection)
 
+	ch, err := amqpConn.connection.Channel()
```

Comment:
maybe we could have a function called CreateChannel in the connection that do this and we pass the result to the constructor as a single parameter